### PR TITLE
Make SIMD level configurable

### PR DESCRIPTION
**Pull request by sekrit-twc**

### CODE REVIEW

File: src/core/kernel/cpulevel.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with VapourSynth; if not, write to the Free Software
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#include <cstring>
```
Comment (Youka) [inline src/core/kernel/cpulevel.cpp]: Shouldn't this be a **C** file?  
(see associated header)

File: src/core/mergefilters.c [hunk 2]
```diff
@@ -455,24 +462,27 @@ static const VSFrameRef *VS_CC maskedMergeGetFrame(int n, int activationReason,
...
+                if (!func) {
+                    if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 1)
+                        func = d->premultiplied ? (yuvhandling ? vs_mask_merge_premul_byte_c : vs_mask_merge_premul_byte_c) : vs_mask_merge_byte_c;
+                    else if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 2)
+                        func = d->premultiplied ? (yuvhandling ? vs_mask_merge_premul_word_c : vs_mask_merge_premul_word_c) : vs_mask_merge_word_c;
+                    else if (d->vi->format->sampleType == stFloat && d->vi->format->bytesPerSample == 4)
+                        func = d->premultiplied ? vs_mask_merge_premul_float_c : vs_mask_merge_float_c;
+                }
+
+                if (!func)
```
Comment (Youka) [inline src/core/mergefilters.c]: This redundancy could be avoided by moving the `continue` as `else` condition into the block above.

File: src/core/mergefilters.c [hunk 3]
```diff
@@ -441,7 +448,7 @@ static const VSFrameRef *VS_CC maskedMergeGetFrame(int n, int activationReason,
                 const uint8_t *maskp = vsapi->getReadPtr((plane && mask23) ? mask23 : mask, d->first_plane ? 0 : plane);
                 uint8_t * VS_RESTRICT dstp = vsapi->getWritePtr(dst, plane);
 
-                void (*func)(const void *, const void *, const void *, void *, unsigned, unsigned, unsigned);
+                void (*func)(const void *, const void *, const void *, void *, unsigned, unsigned, unsigned) = 0;
```
Comment (Youka) [inline src/core/mergefilters.c]: `0` -> `NULL` (as seen in other parts of this PR)

File: src/core/mergefilters.c [hunk 4]
```diff
@@ -263,28 +265,30 @@ static const VSFrameRef *VS_CC mergeGetFrame(int n, int activationReason, void *
...
+                if (!func) {
+                    if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 1)
+                        func = vs_merge_byte_c;
+                    else if (d->vi->format->sampleType == stInteger && d->vi->format->bytesPerSample == 2)
+                        func = vs_merge_word_c;
+                    else if (d->vi->format->sampleType == stFloat && d->vi->format->bytesPerSample == 4)
+                        func = vs_merge_float_c;
+                }
+
+                if (!func)
```
Comment (Youka) [inline src/core/mergefilters.c]: This redundancy could be avoided by moving the `continue` as `else` condition into the block above.  
(happens again)

File: src/core/mergefilters.c [hunk 5]
```diff
@@ -263,28 +265,30 @@ static const VSFrameRef *VS_CC mergeGetFrame(int n, int activationReason, void *
                 const uint8_t *srcp2 = vsapi->getReadPtr(src2, plane);
                 uint8_t * VS_RESTRICT dstp = vsapi->getWritePtr(dst, plane);
 
-                void (*func)(const void *, const void *, void *, union vs_merge_weight, unsigned);
+                void (*func)(const void *, const void *, void *, union vs_merge_weight, unsigned) = 0;
```
Comment (Youka) [inline src/core/mergefilters.c]: `0` -> `NULL` (as seen in other parts of this PR)  
(happens again)

File: src/core/kernel/cpulevel.cpp (hunk 1 above)
Comment (dubhater) [reply src/core/kernel/cpulevel.cpp]: What for?

**Comment by stuxcrystal** [comment]
You did not just merge a PR with open Code-Reviews, did you?

File: src/core/kernel/cpulevel.cpp (hunk 1 above)
Comment (Youka) [reply src/core/kernel/cpulevel.cpp]: * Header defines functions in C context
* File contains solely C code (f.e. `strcmp`)

**Comment by myrsloik** [comment]
> You did not just merge a PR with open Code-Reviews, did you?

Sure did. No review of value was lost.

**Comment by myrsloik** [comment]
> You did not just merge a PR with open Code-Reviews, did you?

What review? I've now blocked @Youka

File: src/core/kernel/cpulevel.cpp (hunk 1 above)
Comment (sekrit-twc) [reply src/core/kernel/cpulevel.cpp]: `core->getCpuLevel()` is a C++ statement...

Comment (Youka) [reply src/core/kernel/cpulevel.cpp]: Oh, i didn't see **VSCore** isn't a struct holding function pointers but a public C++ class, sorry. Was already wondering why these thin wrapper functions exist anyway.

Found this C & C++ mix strange. `!strcmp(...)` always looks like the opposite expectation, using [std::string == operator](https://en.cppreference.com/w/cpp/string/basic_string/operator_cmp) looks more logical and already stores string length for faster compare.

**Pull request merged**